lib/Doctrine/ORM/Internal/Hydration/ScalarColumnHydrator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\ORM\Internal\Hydration;
+
+use PDO;
+
+use function reset;
+use function array_keys;
+use function array_column;
+
+/**
+ * The ScalarColumnHydrator is extended ArrayHydrator which produces a flat array single column "graph"
+ * that is often (not always) interchangeable with the corresponding object graph for read-only access.
+ */
+class ScalarColumnHydrator extends ArrayHydrator
```

Comment:
@ostrolucky I don't get why we extend from `ArrayHydrator` here… the implementation I suggested should be more simple, shouldn't it?

Reply:
Decoupling from ArrayHydrator would require additional work, hence why I didn't do it before. I can't tell how much more work would it require.

Reply:
I am trying to decouple things, so far things are good, after some time I will push latest changes, Which you guys can review.

Reply:
How `fetchOne` should behave ? 


```php
    public function fetchOne()
    {
        $current = current($this->resultSet);
        next($this->resultSet);

        return $current;
    }
```

is this correct ? @greg0ire

Reply:
Yes that does look correct 👍

Reply:
Simplified implementation accordingly